Mathlib/CategoryTheory/Adjunction/Opposites.lean

```diff
@@ -0,0 +1,424 @@
+/-
+Copyright (c) 2020 Bhavik Mehta. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Bhavik Mehta, Thomas Read, Andrew Yang
+
+! This file was ported from Lean 3 source module category_theory.adjunction.opposites
+! leanprover-community/mathlib commit f3ee4628e2dc737653af924c41fa681abc2a4f4a
+! Please do not edit these lines, except to modify the commit id
+! if you have ported upstream changes.
+-/
+import Mathlib.CategoryTheory.Adjunction.Basic
+import Mathlib.CategoryTheory.Yoneda
+import Mathlib.CategoryTheory.Opposites
+
+/-!
+# Opposite adjunctions
+
+This file contains constructions to relate adjunctions of functors to adjunctions of their
+opposites.
+These constructions are used to show uniqueness of adjoints (up to natural isomorphism).
+
+## Tags
+adjunction, opposite, uniqueness
+-/
+
+
+open CategoryTheory
+
+universe v₁ v₂ u₁ u₂
+
+-- morphism levels before object levels. See note [CategoryTheory universes].
+variable {C : Type u₁} [Category.{v₁} C] {D : Type u₂} [Category.{v₂} D]
+
+namespace CategoryTheory.Adjunction
+
+/-- If `G.op` is adjoint to `F.op` then `F` is adjoint to `G`. -/
+@[simps!]
+def adjointOfOpAdjointOp (F : C ⥤ D) (G : D ⥤ C) (h : G.op ⊣ F.op) : F ⊣ G :=
+  Adjunction.mkOfHomEquiv {
+    homEquiv := fun {X Y} =>
+      ((h.homEquiv (Opposite.op Y) (Opposite.op X)).trans (opEquiv _ _)).symm.trans
+        (opEquiv _ _)
+    homEquiv_naturality_left_symm := by
```

Comment:
I would have hope there was a porting note here noting the regression that this proof is needed at all!

Reply:
Done!